nixos/modules/services/matrix/synapse.nix

```diff
@@ -1022,15 +1096,26 @@ in {
           This is done by default unless you manually configure either of those settings.
         '';
       }
-    ];
+    ]
+    ++ (map (listener: {
+      assertion = listener.path == null || (listener.bind_addresses == null && listener.port == null && listener.tls == null);
```

Comment:
This accepts a listener with everything being `null` whcih seems wrong:

```
{    path = null;
    bind_addresses = null;
    port = null;
    tls = null;
}
```

Reply:
I'm stuck at fixing the assertion. See the last fixup commit.

```
trace: null

[...]

       error:
       Failed assertions:
       - Listener configured with UNIX domain socket () ignores the following options:
         - bind_addresses=127.0.0.1 ::1
         - port=8448
         - tls=1
`

The `listener.path` traces as `null`, and the assertion checks for `listener.path != null`.